### PR TITLE
Integration tests fixes

### PR DESCRIPTION
**Pull request by neykov**
Either fixes or marks integration tests as Broken so the CI integration test runs reliably.

### CODE REVIEW

File: core/src/test/java/org/apache/brooklyn/core/entity/ApplicationLifecycleStateTest.java [hunk 1]
```diff
@@ -201,6 +227,163 @@ public void testStartsThenChildFailsButWithQuorumCausesAppToStayHealthy() throws
         assertUpAndRunningEventually(app);
     }
 
+    /**
+     * Sensor value does not match the order of events. For example events are running, starting,
```
Comment (aledsage) [inline core/src/test/java/org/apache/brooklyn/core/entity/ApplicationLifecycleStateTest.java]: Maybe reword the javadoc - I think this refers to the situation before you made the fix.

How about:
```
    /**
      * Tests concurrent modifications to a sensor, asserting that the last notification the subscribers 
      * receives equals the last value that sensor has.
      * 
      * Prior to this being fixed (see https://github.com/apache/brooklyn-server/pull/622), it caused 
      * problems in ComputeServiceIndicatorsFromChildrenAndMembers: it saw a child transition 
      * from "running" to "starting", and thus emitted the on-fire event for the parent entity. As asserted
      * by this test, the enricher should now always receive the events in the correct order (e.g. "starting",
      * "running").
      */
```

File: core/src/test/java/org/apache/brooklyn/core/entity/ApplicationLifecycleStateTest.java [hunk 2]
```diff
@@ -201,6 +227,163 @@ public void testStartsThenChildFailsButWithQuorumCausesAppToStayHealthy() throws
...
+        }
+
+        @Override
+        public <T> void emit(Sensor<T> sensor, Object val) {
+            super.emit(sensor, val);
+        }
+    }
+
+    /**
+     * The deduplication logic in AbstractEnricher does not work for parallel invocations.
```
Comment (aledsage) [inline core/src/test/java/org/apache/brooklyn/core/entity/ApplicationLifecycleStateTest.java]: javadoc no longer correct, now that you've fixed it. How about:
```
      * The deduplication logic in AbstractEnricher previously did not work for parallel invocations.
      * It used to do a get and then a compare, so another thread could change the value between
      * those two operations.
```

File: core/src/test/java/org/apache/brooklyn/core/mgmt/rebind/RebindTestUtils.java [hunk 3]
```diff
@@ -238,7 +238,6 @@ public LocalManagementContext buildUnstarted() {
                     classLoader);
             ((RebindManagerImpl) unstarted.getRebindManager()).setPeriodicPersistPeriod(persistPeriod);
             unstarted.getRebindManager().setPersister(newPersister, PersistenceExceptionHandlerImpl.builder().build());
-            unstarted.getHighAvailabilityManager().disabled();
```
Comment (aledsage) [inline core/src/test/java/org/apache/brooklyn/core/mgmt/rebind/RebindTestUtils.java]: Not sure about the implications of removing this. It seems reasonable that we execute the HA code-path in our tests, so I think this is fine. I take it that some of our integration tests assumd that it's enabled?

Comment (neykov) [inline core/src/test/java/org/apache/brooklyn/core/mgmt/rebind/RebindTestUtils.java]: I added this recently in https://github.com/apache/brooklyn-server/pull/607, but no longer needed after the changes in https://github.com/apache/brooklyn-server/pull/617 which make the HA state configurable.

**Comment by neykov** [comment]
Thanks @aledsage. Updated the javadocs. Waiting for the new build to finish and will merge.

**Comment by aledsage** [comment]
@neykov I ran this locally - it hung. Doing `jstack` showed a deadlock! Looks like this is caused by the synchronization changes in `AttributeMap.update`.
```
Found one Java-level deadlock:
=============================
"brooklyn-execmanager-P37YeS8X-2":
  waiting to lock monitor 0x00007ff7f7c8be98 (object 0x00000007b482f468, a org.apache.brooklyn.core.mgmt.internal.EntityManagementSupport),
  which is held by "main"
"main":
  waiting to lock monitor 0x00007ff7f847fe08 (object 0x00000007b482f880, a java.util.Collections$SynchronizedMap),
  which is held by "brooklyn-execmanager-P37YeS8X-2"

Java stack information for the threads listed above:
===================================================
"brooklyn-execmanager-P37YeS8X-2":
        at org.apache.brooklyn.core.mgmt.internal.EntityManagementSupport.getSubscriptionContext(EntityManagementSupport.java:363)
        - waiting to lock <0x00000007b482f468> (a org.apache.brooklyn.core.mgmt.internal.EntityManagementSupport)
        at org.apache.brooklyn.core.entity.AbstractEntity$BasicSubscriptionSupport.getSubscriptionContext(AbstractEntity.java:1431)
        at org.apache.brooklyn.core.entity.AbstractEntity$BasicSensorSupport.emitInternal(AbstractEntity.java:1215)
        at org.apache.brooklyn.core.entity.AbstractEntity.emitInternal(AbstractEntity.java:1996)
        at org.apache.brooklyn.core.sensor.AttributeMap.update(AttributeMap.java:136)
        - locked <0x00000007b482f880> (a java.util.Collections$SynchronizedMap)
        at org.apache.brooklyn.core.entity.AbstractEntity$BasicSensorSupport.set(AbstractEntity.java:1114)
        at org.apache.brooklyn.core.enricher.AbstractEnricher.emit(AbstractEnricher.java:137)
        at org.apache.brooklyn.enricher.stock.AbstractTransformer.onEvent(AbstractTransformer.java:148)
        at org.apache.brooklyn.core.mgmt.internal.LocalSubscriptionManager$1.run(LocalSubscriptionManager.java:276)
        at org.apache.brooklyn.util.concurrent.CallableFromRunnable.call(CallableFromRunnable.java:44)
        at org.apache.brooklyn.util.core.task.BasicExecutionManager$SubmissionCallable.call(BasicExecutionManager.java:529)
        at org.apache.brooklyn.util.core.task.SingleThreadedScheduler$1.call(SingleThreadedScheduler.java:116)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
"main":
        at java.util.Collections$SynchronizedMap.get(Collections.java:2584)
        - waiting to lock <0x00000007b482f880> (a java.util.Collections$SynchronizedMap)
        at org.apache.brooklyn.core.sensor.AttributeMap.getValue(AttributeMap.java:204)
        at org.apache.brooklyn.core.sensor.AttributeMap.getValue(AttributeMap.java:210)
        at org.apache.brooklyn.core.entity.AbstractEntity$BasicSensorSupport.get(AbstractEntity.java:1089)
        at org.apache.brooklyn.core.entity.AbstractEntity.getAttribute(AbstractEntity.java:996)
        at org.apache.brooklyn.core.mgmt.internal.LocalSubscriptionManager.subscribe(LocalSubscriptionManager.java:147)
        - locked <0x00000007b4777818> (a org.apache.brooklyn.core.mgmt.internal.LocalSubscriptionManager)
        at org.apache.brooklyn.core.mgmt.internal.QueueingSubscriptionManager.startDelegatingForSubscribing(QueueingSubscriptionManager.java:81)
        - locked <0x00000007b482f518> (a org.apache.brooklyn.core.mgmt.internal.QueueingSubscriptionManager)
        at org.apache.brooklyn.core.mgmt.internal.EntityManagementSupport.onManagementStarting(EntityManagementSupport.java:186)
        - locked <0x00000007b482f468> (a org.apache.brooklyn.core.mgmt.internal.EntityManagementSupport)
        at org.apache.brooklyn.core.mgmt.internal.LocalEntityManager$2.apply(LocalEntityManager.java:392)
        at org.apache.brooklyn.core.mgmt.internal.LocalEntityManager$2.apply(LocalEntityManager.java:348)
        at org.apache.brooklyn.core.mgmt.internal.LocalEntityManager.recursively(LocalEntityManager.java:619)
        at org.apache.brooklyn.core.mgmt.internal.LocalEntityManager.manageRecursive(LocalEntityManager.java:413)
        at org.apache.brooklyn.core.mgmt.internal.LocalEntityManager.manage(LocalEntityManager.java:307)
        at org.apache.brooklyn.core.mgmt.internal.LocalEntityManager.createEntity(LocalEntityManager.java:153)
        at org.apache.brooklyn.core.entity.AbstractEntity.addChild(AbstractEntity.java:710)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.brooklyn.core.objs.proxy.EntityProxyImpl.invoke(EntityProxyImpl.java:201)
        at com.sun.proxy.$Proxy198.addChild(Unknown Source)
        at org.apache.brooklyn.entity.software.base.EntitySshToolTest.testCustomSshToolClassConfiguredOnBrooklynProperties(EntitySshToolTest.java:85)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
        at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
        at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
        at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
        at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
        at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
        at org.testng.TestRunner.privateRun(TestRunner.java:756)
        at org.testng.TestRunner.run(TestRunner.java:610)
        at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
        at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
        at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
        at org.testng.SuiteRunner.run(SuiteRunner.java:289)
        at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
        at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
        at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
        at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
        at org.testng.TestNG.runSuites(TestNG.java:1133)
        at org.testng.TestNG.run(TestNG.java:1104)
        at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:132)
        at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeMulti(TestNGDirectoryTestSuite.java:193)
        at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:94)
        at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:147)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)

Found 1 deadlock.
```

**Comment by neykov** [comment]
Thanks for the analysis @aledsage. I split the changes causing the failure in a separate PR (https://github.com/apache/brooklyn-server/pull/641) so we can merge this one independently.
